components/builder-graph/src/acyclic_package_graph.rs

```diff
@@ -234,10 +247,163 @@ impl PackageGraphTrait for AcyclicPackageGraph {
                 connected_comp: connected_components(&self.graph),
                 is_cyclic:      is_cyclic_directed(&self.graph), }
     }
+
+    // compute_build for AcyclicGraph
+    // Process
+    // 1) Take kernel of packages, and recursively expand it over reverse build/runtime deps
+    //    creating a new sub-graph.
+    // 2) Filter unbuildable nodes in subgraph, flag them as directly
+    //    (no plan connection, autobuilds off) unbuildable
+    // 3) Use the unbuildable packages as kernel, expand it over the subgraph to
+    //    create a list of all unbuildable packages.
+    // 4) Iterate over all unbuildable packages.
+    //    Flag new entries as indirectly unbuildable.
+    //    Remove that package from the sub-graph
+    // 5) Walk the remaining sub-graph, collecting dependencies that are external to
+    //    the sub-graph.
+    // 6) Create a PackageBuildManifestGraph using the information generated in 1-5.
+    fn compute_build(&self,
+                     touched: &[PackageIdentIntern],
+                     unbuildable: &dyn Unbuildable)
+                     -> Result<PackageBuildManifest> {
+        let mut rebuild_graph: DiGraphMap<PackageIdentIntern, EdgeType> = DiGraphMap::new();
+        let mut unbuildable_map: HashMap<PackageIdentIntern, UnbuildableReason> = HashMap::new();
+        let mut all_external_dependencies: HashSet<PackageIdentIntern> = HashSet::new();
+
+        let mut worklist: VecDeque<PackageIdentIntern> =
+            VecDeque::from_iter(touched.iter().copied());
+        let mut seen: HashSet<PackageIdentIntern> = HashSet::from_iter(touched.iter().copied());
+
+        // Starting with our 'touched' set, walk the acyclic graph adding all
+        // reverse dependencies to a new sub-graph.
+        //
+        // Error case: A package is missing from the graph
+        //   * This package is flagged as missing in the unbuildables list.
+        while !worklist.is_empty() {
+            // Assumption: package is a short ident
+            let package: PackageIdentIntern = worklist.pop_back().unwrap(); // List is not empty
+
+            if let Some((_, node_index)) = self.package_map.get(&package.to_string()) {
+                // We reverse the sense of dependency edges when building the graph, so 'outgoing'
+                // edges are really pointing to packages that depend on us
+                for neighbor in self.graph
+                                    .neighbors_directed(*node_index, Direction::Outgoing)
+                {
+                    let short_ident =
```

Comment:
Similarly, naming this "dependency_ident" (or similar) would be better for readability... naming after its role / what it does, rather than what it concretely is.